tests/ui/non_canonical_clone_impl.stderr

```diff
@@ -29,13 +29,19 @@ LL | |     }
    | |_____^ help: change this to: `{ *self }`
 
 error: unnecessary implementation of `clone_from` on a `Copy` type
-  --> tests/ui/non_canonical_clone_impl.rs:85:5
+  --> tests/ui/non_canonical_clone_impl.rs:90:5
    |
 LL | /     fn clone_from(&mut self, source: &Self) {
 LL | |         source.clone();
 LL | |         *self = source.clone();
 LL | |     }
    | |_____^ help: remove it
 
-error: aborting due to 4 previous errors
+error: non-canonical implementation of `clone` on a `Copy` type
+  --> tests/ui/non_canonical_clone_impl.rs:120:5
+   |
+LL |     span
+   |     ^^^^ help: change this to: `{ *self }`
```

Comment:
We shouldn't warn here. Most likely you need to change this to an `||`

https://github.com/rust-lang/rust-clippy/blob/64be449ad76fc9cb614e11ca9f758bc0c64b30d4/clippy_lints/src/non_canonical_impls.rs#L132